added handling for multiple transcodig profiles

umar8hassan opened the pull request:
### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/7039

### Description (What does it do?)
Added handling for multiple media profiles in OVS

### Testing
This will be tested along with `https://github.com/mitodl/ocw-studio/pull/2421` after publishing.

Anyone interested in testing it locally can do so using the `ocw-studio` repository, as follows:
1. switch to this branch `umar/7039-update-transcoding-app` in the `ol-django` repository.
2. run the following: 
```
docker-compose build
docker-compose run --rm -ti shell bash

# inside shell
uv build --package mitol-django-transcoding
```
3. check that you have a `mitol_django_transcoding-xxxx.x.xx.tar.gz` file in `dist/`
4. Move `list/mitol_django_transcoding-xxxx.x.xx.tar.gz` to ocw-studio repository.
5.  switch to branch `umar/6717-refactor-transcoding-job-code` in ocw-studio.
6. Spin up containers after rebuilding: `docker-compose up -d --build`
7. Test the transcoding functionality using the instructions in readme or [here](https://github.com/mitodl/ocw-studio/pull/469)

mbertrand reviewed (commented):
Code looks good, is there a different ocw-studio branch other than `umar/6717-refactor-transcoding-job-code` that I should use to test this?  That one was last updated a few weeks ago and doesn't include the new settings `VIDEO_S3_UPLOAD_PREFIX ` and `VIDEO_S3_TRANSCODE_BUCKET`

umar8hassan commented:
> Code looks good, is there a different ocw-studio branch other than `umar/6717-refactor-transcoding-job-code` that I should use to test this? That one was last updated a few weeks ago and doesn't include the new settings `VIDEO_S3_UPLOAD_PREFIX ` and `VIDEO_S3_TRANSCODE_BUCKET`

@mbertrand I have updated this branch (added missing settings here as they are common) and rebased `umar/6717-refactor-transcoding-job-code` for updates. This is ready for testing with `ocw-studio`.

mbertrand reviewed (approved):
The transcode job succeeded, and when I changed the values of VIDEO_S3_TRANSCODE_BUCKET and VIDEO_S3_TRANSCODE_PREFIX that was reflected in the job output.
